python/ray/ml/predictors/integrations/huggingface/huggingface_predictor.py

```diff
@@ -0,0 +1,209 @@
+
+        The checkpoint is expected to be a result of ``HuggingFaceTrainer``.
+
+        Args:
+            checkpoint: The checkpoint to load the model and
+                preprocessor from. It is expected to be from the result of a
+                ``HuggingFaceTrainer`` run.
+            model: Either a ``transformers.PreTrainedModel`` class
+                (eg. ``AutoModelForCausalLM``), or a PyTorch model to load the
+                weights to. This should be the same model used for training.
+            **pretrained_model_kwargs: Any kwargs to pass to the
+                ``model.from_pretrained()`` call. Only used if
+                ``model`` is a ``PreTrainerModel`` class.
+        """
+        with checkpoint.as_directory() as checkpoint_path:
+            preprocessor_path = os.path.join(checkpoint_path, PREPROCESSOR_KEY)
+            if os.path.exists(preprocessor_path):
+                with open(preprocessor_path, "rb") as f:
+                    preprocessor = cpickle.load(f)
+            else:
+                preprocessor = None
+            if isinstance(model, torch.nn.Module):
+                state_dict = torch.load(
+                    os.path.join(checkpoint_path, WEIGHTS_NAME), map_location="cpu"
+                )
+                model = load_torch_model(saved_model=state_dict, model_definition=model)
+            else:
+                model = model.from_pretrained(
+                    checkpoint_path, **pretrained_model_kwargs
+                )
+            training_args_path = os.path.join(checkpoint_path, TRAINING_ARGS_NAME)
+            if os.path.exists(training_args_path):
+                with open(training_args_path, "rb") as f:
+                    training_args = torch.load(f, map_location="cpu")
+            else:
+                training_args = None
+        return HuggingFacePredictor(
+            model=model, preprocessor=preprocessor, training_args=training_args
+        )
+
+    def _convert_to_tensor(
+        self,
+        data: pd.DataFrame,
+        feature_columns: Optional[
+            Union[List[str], List[List[str]], List[int], List[List[int]]]
+        ] = None,
+        dtypes: Optional[torch.dtype] = None,
+        unsqueeze: bool = False,
+    ) -> Dict[str, torch.Tensor]:
+        if not feature_columns:
+            feature_columns = data.columns
+
+        # HF-supported format
+        if not isinstance(feature_columns, dict):
+            feature_columns = {column: [column] for column in feature_columns}
+
+        return super()._convert_to_tensor(
+            data, feature_columns=feature_columns, dtypes=dtypes, unsqueeze=unsqueeze
+        )
+
+    def _predict(self, tensor: Dict[str, torch.Tensor]) -> pd.DataFrame:
+        if self.training_args:
+            self.training_args.local_rank = -1
+        trainer = HFTrainer(model=self.model, args=self.training_args)
+        dataset = HFIterableDatasetWithLen([tensor], 1)
```

Comment:
Can we follow up with the HF team on the right way to do prediction? It seems weird to need to instantiate a Trainer to do inference.